src/widgets/survey/model/hooks/useFlowStateActions.ts

```diff
@@ -128,11 +137,22 @@ export function useFlowStateActions({
     }
   }
 
-  function completeByTimer() {
+  function completeByTimer(): TimerCallbackFeedback {
```

Comment:
I am not sure why it should return something related to timer callback when it is not defined as a callback (it is defined just as an exportable function) and has nothing to do with how it is going to be used.

Reply:
Otherwise, we face with race-condition when callback is fired during uploading on Interim screen and then we switch to the Finish screen and doing the same actions (autocompletion as well).

Reply:
Resolved by Postponer